packages/frontend/features/swap-tokens/SwapOptions.tsx

```diff
@@ -55,36 +55,20 @@ const SwapOptions = () => {
     setAmountToSwap(canSwapAmount.toString());
   };
 
-  const getAmountToReceive = async () => {
-    const { uniswapFactory } = contracts;
-
+  const updateResultAmount = async () => {
     setAmountToReceive(null);
 
-    const fromToken = COINS[fromTokenStr];
-    const toToken = COINS[toTokenStr];
-
-    const amountWei = ethers.utils.parseUnits(amountToSwap, fromToken.decimals);
-
-    const receivedWei = await useSwapResult(
+    const received = await useGetAmountToReceive(
       signer,
-      uniswapFactory,
-      fromToken.address,
-      toToken.address,
-      amountWei
-    );
-    // Minus 0.135% in fees if colalteral, else add 0.135%
-    const receivedFixed =
-      thingToSwap === "debt"
-        ? receivedWei.mul(100135).div(100000)
-        : receivedWei.mul(99865).div(100000);
-
-    const received = ethers.utils.formatUnits(
-      receivedFixed.toString(),
-      toToken.decimals
+      contracts.uniswapFactory,
+      thingToSwap,
+      COINS[fromTokenStr],
+      amountToSwap,
+      COINS[toTokenStr]
```

Comment:
you should not be passing in any of these things except for:

- `thingToSwap`
- `fromTokenStr`
- `toTokenStr`
- `amountToSwap`

Everything else can be retrieved inside the hook. Avoid using the state containers inside components, it should be inside the custom hook instead. I can clarify by refactoring this myself if you'd like. Or we could pair.